reactive/kotlinx-coroutines-reactive/test/ReactiveStreamTckTest.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.reactive
+
+import kotlinx.coroutines.*
+import org.junit.*
+import org.junit.runner.*
+import org.junit.runners.*
+import org.reactivestreams.*
+import org.reactivestreams.tck.*
+
+@RunWith(Parameterized::class)
+class ReactiveStreamTckTest(
+    private val dispatcher: Dispatcher
+) : PublisherVerification<Long>(TestEnvironment()) {
+
+    enum class Dispatcher(val dispatcher: CoroutineDispatcher)  {
+        DEFAULT(Dispatchers.Default),
+        UNCONFINED(Dispatchers.Unconfined)
+    }
+
+    private val scope = CoroutineScope(dispatcher.dispatcher)
+
+    companion object {
+        @Parameterized.Parameters(name = "{0}")
+        @JvmStatic
+        fun params(): Collection<Array<Any>> = Dispatcher.values().map { arrayOf<Any>(it) }
```

Comment:
You don't need the nested array here. See "Tests with single parameter" on https://junit.org/junit4/javadoc/4.12/org/junit/runners/Parameterized.html

Reply:
Good to know. Thanks!